tests/main/parallel-install-services/task.yaml

```diff
@@ -0,0 +1,47 @@
+summary: Checks for parallel installation of sideloaded snaps containing services
+
+prepare:
+    snap set system experimental.parallel-instances=true
+
+restore:
+    snap set system experimental.parallel-instances=null
+
+execute: |
+    #shellcheck source=tests/lib/snaps.sh
+    . "$TESTSLIB"/snaps.sh
+
+    install_local test-snapd-service
+
+    #shellcheck source=tests/lib/dirs.sh
+    . "$TESTSLIB"/dirs.sh
+
+    for instance in foo longname; do
+        echo "Install a snap as instance named test-snapd-service_$instance"
+        expected="^test-snapd-service_$instance 1.0 installed\$"
+        install_local_as test-snapd-service "test-snapd-service_$instance" | MATCH "$expected"
+
+        test -d "$SNAP_MOUNT_DIR/test-snapd-service_$instance/x1"
+
+        test -f /etc/systemd/system/snap.test-snapd-service_$instance.test-snapd-service.service
+        test -f /etc/systemd/system/snap.test-snapd-service_$instance.test-snapd-other-service.service
+        systemctl is-active snap.test-snapd-service_$instance.test-snapd-service.service
+        systemctl is-active snap.test-snapd-service_$instance.test-snapd-other-service.service
+    done
+
+    echo "All snaps are listed"
+    snap list | MATCH '^test-snapd-service '
+    snap list | MATCH '^test-snapd-service_foo '
+    snap list | MATCH '^test-snapd-service_longname '
+
+    echo "Removing one instance does not remove services from other instances"
+    snap remove test-snapd-service_foo
+    ! test -f /etc/systemd/system/snap.test-snapd-service_foo.test-snapd-service.service
+    test -f /etc/systemd/system/snap.test-snapd-service_longname.test-snapd-service.service
+    test -f /etc/systemd/system/snap.test-snapd-service.test-snapd-service.service
```

Comment:
We could (for good measure) test here that the services are still running too (i.e. that they did not get accidentally  stopped)

Reply:
This is done just below now.